task-1-solution/FirstTaskSolution/app/src/main/res/values/dimens.xml

```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <!-- TextSize -->
+    <dimen name="smallTextSize">14sp</dimen>
+    <dimen name="normalTextSize">16sp</dimen>
+    <dimen name="largeTextSize">20sp</dimen>
+    <dimen name="hugeTextSize">64sp</dimen>
+
+    <!-- VerticalSpacing -->
+
+    <!-- SmallVerticalSpacing -->
+    <dimen name="small100VerticalSpacing">4dp</dimen>
+    <dimen name="small200VerticalSpacing">8dp</dimen>
+    <dimen name="small300VerticalSpacing">12dp</dimen>
+    <dimen name="small400VerticalSpacing">16dp</dimen>
+    <dimen name="small500VerticalSpacing">20dp</dimen>
+    <dimen name="small600VerticalSpacing">24dp</dimen>
+    <dimen name="small700VerticalSpacing">28dp</dimen>
+    <dimen name="small800VerticalSpacing">32dp</dimen>
+    <dimen name="small900VerticalSpacing">36dp</dimen>
+    <!-- MediumVerticalSpacing -->
+    <dimen name="medium100VerticalSpacing">40dp</dimen>
+    <dimen name="medium200VerticalSpacing">44dp</dimen>
+    <dimen name="medium300VerticalSpacing">48dp</dimen>
+    <dimen name="medium400VerticalSpacing">52dp</dimen>
+    <dimen name="medium500VerticalSpacing">56dp</dimen>
+    <dimen name="medium600VerticalSpacing">64dp</dimen>
+    <dimen name="medium700VerticalSpacing">72dp</dimen>
+    <dimen name="medium800VerticalSpacing">80dp</dimen>
+    <dimen name="medium900VerticalSpacing">88dp</dimen>
+    <!-- LargeSizeVerticalSpacing -->
+    <dimen name="large100VerticalSpacing">92dp</dimen>
+    <dimen name="large200VerticalSpacing">96dp</dimen>
+    <dimen name="large300VerticalSpacing">100dp</dimen>
+    <dimen name="large400VerticalSpacing">104dp</dimen>
+    <dimen name="large500VerticalSpacing">108dp</dimen>
+    <dimen name="large600VerticalSpacing">112dp</dimen>
+    <dimen name="large700VerticalSpacing">116dp</dimen>
+    <dimen name="large800VerticalSpacing">120dp</dimen>
+    <dimen name="large900VerticalSpacing">124dp</dimen>
```

Comment:
Eline sağlık düzenli olmuş ama layout içerisinde bu büyüklüklerde spacing kullanmak uygun değil. Küçük boyutlu ekranlarda bozulmalara sebebiyet verebilir.

En büyük spacing değeri `32dp` olabilir. İstinai durumlarda `48dp`'ye çıkabilir.

Aralarda ekstradan boşluk vermeden `constraint`'leri ayarlayabilmek `ConstraintLayout`'un bize sağladığı kolaylıklardan zaten. Bunu kullanmazsak, `ConstraintLayout`'dan tam faydalanmamış oluruz.

O yüzden bu task kapsamında `48dp` üzerindekileri kullanmayalım. Bundan sonraki task'larda da mümkün mertebe daha da düşük boşluklar kullanalım.

Konuştuğumuz sadece spacingler için. `width` ve `height` değerleri için bu kural daha fazla esnetilebilir.